### ISSUE TITLE
NumberBox doesn't trigger ValueChanged if X is clicked

### ISSUE THREAD

**Issue by DraTeots** [amwx/FluentAvalonia#489]
**Describe the bug**

When number is being changed by mouse, there is a small [x] button. If the button is pressed, it erases the text and sets `Value` to NaN, but `ValueChanged` event is not triggered. 

xaml control definitions: 

```xaml
<controls:NumberBox Grid.Row="0" Name="CekNumberBox" Description="    cek" ValueChanged="AnyNumberBox_ValueChanged" ValidationMode="InvalidInputOverwritten" SimpleNumberFormat="##0." Margin="0,0,5,0" SpinButtonPlacementMode="Compact"></controls:NumberBox>
```

**Screenshots**

![image](https://github.com/amwx/FluentAvalonia/assets/4732545/2c2f8888-0665-46ce-9836-c310e7ae1d2f)

**Desktop/Platform (please complete the following information):**
 - OS: Linux 6.2.0-34-generic #22.04.1-Ubuntu
 - FluentAvalonia Version 2.0.4
 - Avalonia Version 11.0.4

**Additional context**

Is it possible to have an option switch off [x] button at all? 

Rationale: [X] appears right where arrows are originally located. So you click arrows to change value with mouse, the control transforms and [x] now right under your mouse. If you misclick, the value is instantly lost and you can't use side arrows anymore or anything to restore a value. Since arrows in the NumberBox control allows to change value with mouse this UX decision is kind of contradicts control intent. 

![image](https://github.com/amwx/FluentAvalonia/assets/4732545/a0bac3f9-ca06-4aa6-b324-4a9bd2c096b8)

**Comment by timunie** [comment]
> Is it possible to have an option to switch off [x] button at all?

Yes, define a Style for TextBox with this selector, but with no clear button inside. Alternatively, re-define the ControlTemplate
https://github.com/amwx/FluentAvalonia/blob/759d12a67f623a15f44c18c10e0e7480a69aa8e1/src/FluentAvalonia/Styling/ControlThemes/FAControls/NumberBoxStyles.axaml#L181C43-L181C54

> image
I prefer to set a MinWidth to controls like this. That should solve the growth when you set focus afaik

**Comment by DraTeots** [comment]
Thank you for pointing to the template and the suggestions!
